### ISSUE TITLE
Add cypress_test e2e testing rule

### ISSUE THREAD

**Issue by Toxicable** [bazelbuild/rules_nodejs#607]
# 🚀 feature request

### Description
A more DX friendly e2e testing rule

### Describe the solution you'd like
`cypress_test` rule with support for ibazel test/run - `cypress open` and bazel test/run - `cypress run`

### Describe alternatives you've considered
`protractor_web_test_suite`

###

### Test
- Cypress tries to touch the Read Only FS during bazel test - https://github.com/cypress-io/cypress/issues/1281
### Run
- Cannot run cypress.run with no cypress.json - https://github.com/cypress-io/cypress/issues/3675
### ibazel
- Cannot specify what specs to run during `cypress open` - https://github.com/cypress-io/cypress/issues/1925

**Comment by Globegitter** [comment]
One way to get around the write issue from bazel's pov is by adding the `--sandbox_writable_path=/ ` but that is of course not great as it renders that the read-only FS useless.

Also what is strange, we started to integrate cypress into our bazel CI setup recently (in a quite "quick and dirty" way) and while I do get the read-only FS error when running `bazel test` locally strangely enough our circleci setup just works:

```
It looks like this is your first time using Cypress: 3.2.0

[19:22:17]  Verifying Cypress can run /home/circleci/.cache/Cypress/3.2.0/Cypress [started]
[19:22:19]  Verified Cypress!       /home/circleci/.cache/Cypress/3.2.0/Cypress [title changed]
[19:22:19]  Verified Cypress!       /home/circleci/.cache/Cypress/3.2.0/Cypress [completed]
```

I am really not sure how that is possible, I can only guess it has something to do with the docker setup we are running there.

**Comment by Toxicable** [comment]
@Globegitter Is Cypress possibly working outside of the Bazel directories?

**Comment by Globegitter** [comment]
Yeah that could be, I am still not sure however what the real difference is between the circleci setup and my local one. But I guess I would just need to spend some time investigating. And while it is a bit worrying the positive aspect about it is we can use it now already in our CI setup.

**Comment by marcus-sa** [comment]
What's the prgoress on this?

**Comment by Toxicable** [comment]
We're still blocked on Cypress.
Check the related issues for any updates

**Comment by yohcop** [comment]
I think the 2 referenced issues on cypress were fixed! :)

**Comment by abhinaba-ghosh** [comment]
@Globegitter , Hi, do you have any documentation or example project to make cypress work with bezel? Any guidance will help!

**Comment by marcus-sa** [comment]
@abhinaba-ghosh there's a PR to add cypress rules to Bazel https://github.com/bazelbuild/rules_nodejs/pull/1930

**Comment by mattem** [comment]
#1930 is now merged and released as part of 2.0.0 so this issue can be closed I think. 
[Example here](https://github.com/bazelbuild/rules_nodejs/tree/2.0.0-rc.3/examples/cypress)
